hugo/content/docs/user_interaction.md

```diff
@@ -0,0 +1,111 @@
+---
+title: 'User Interaction'
+Weight: 900
+---
+
+## Interacting with a Sprotty Diagram
+Sprotty offers multiple mouse and keyboard interactions by default listed in the following tables.
+The `CTRL` key in the following is equal to the `CMD` key on Mac.
+
+| Mouse       
+| ----------------- | ---- |
+| left-click:       | toggle selection |
+| CTRL-click:       | add to/remove from selection |
+| left-click : drag | move selection (when on selected element) |
+| left-click : drag | pan viewport (when on diagram background) |
+| mouse wheel:    | zoom |
+
+| Touchpad
+| --------------------- | --- |
+|click                  | toggle selection |
+|drag                   | move selection (when on selected element) |
+|two-finger pan up/down | zoom |
+|zoom                   | zoom |
+
+| Keys
+| ---------- | --- |
+|CTRL-SHIFT-A| select all |
+|CTRL-SHIFT-C| center selection, or if nothing is selected center the entire diagram |
+|CTRL-SHIFT-E| export diagram to SVG |
+|CTRL-SHIFT-F| zoom selection to fill the entire canvas, or if nothing is selected zoom the entire diagram |
+|CTRL-Z      | undo |
+|CTRL-SHIFT-Z| redo |
+
+## Creating Custom Interaction
+
+### Buttons and Button Handlers
+The easiest way to create custom interactions is through buttons. Buttons in Sprotty work similarly to the other model elements. 
+As in the other examples with other types of nodes, we first need to define the model and view class and define its type ID. in addition, we also need to define a button handler of type `IButtonHandler`.
+
+```Typescript
+const container = new ContainerModule((bind, unbind, isBound, rebind) => { 
+        const context = { bind, unbind, isBound, rebind };
+        configureModelElement(context, 'graph', SGraph, SGraphView);
+        configureModelElement(context, 'button:custom', SButton, CustomButtonView);
+
+        configureButtonHandler({bind, isBound}, 'button:custom', CustomButtonHandler);
+
+        configureViewerOptions(context, {
+            needsClientLayout: true,
+            baseDiv: 'sprotty'
+        });
+
+    });
+```
+
+A button handler is a simple injectable class with a `buttonPressed(button: SButton): Action[]` method. 
+returned actions are passed to the `ActionDispatcher` to be handled there. 
+```Typescript
+@injectable()
+export class CustomButtonHandler implements IButtonHandler {
+    buttonPressed(button: SButton): Action[] {
+        alert('button on' + button.parent.id + ' pressed');
+        return [];
+    }
+}
+```
+
+### Mouse and Keyboard Listeners
+Sprotty also offers the ability to attach mouse and keyboard listeners by registering `MouseListener` or `KeyListener`.
+This can be simply done by binding the custom listener to the respective listener type in your DI-container like this:
```

Comment:
```suggestion
Sprotty also offers the ability to attach mouse and keyboard listeners reacting to events of a Sprotty diagram.
This can be simply done by binding custom implementations of `MouseListener` or `KeyListener` as follows:
```